ibtk/src/solvers/impls/CCPoissonPointRelaxationFACOperator.cpp

```diff
@@ -65,96 +65,184 @@
 
 // FORTRAN ROUTINES
 #if (NDIM == 2)
-#define GS_SMOOTH_FC IBTK_FC_FUNC(gssmooth2d, GSSMOOTH2D)
-#define RB_GS_SMOOTH_FC IBTK_FC_FUNC(rbgssmooth2d, RBGSSMOOTH2D)
-#define VC_CELL_GS_SMOOTH_FC IBTK_FC_FUNC(vccellgssmooth2d, VCCELLGSSMOOTH2D)
-#define VC_CELL_RB_GS_SMOOTH_FC IBTK_FC_FUNC(vccellrbgssmooth2d, VCCELLRBGSSMOOTH2D)
+#define CONST_DC_GS_SMOOTH_FC IBTK_FC_FUNC(const_dc_gssmooth2d, CONST_DC_GSSMOOTH2D)
+#define CONST_DC_RB_GS_SMOOTH_FC IBTK_FC_FUNC(const_dc_rbgssmooth2d, CONST_DC_RBGSSMOOTH2D)
+#define VAR_D_CONST_C_GS_SMOOTH_FC IBTK_FC_FUNC(var_d_const_c_gssmooth2d, VAR_D_CONST_C_GSSMOOTH2D)
```

Comment:
Suggest just using variable C and variable D to the VC implementation, instead of adding new specialized Fortran kernels for the 3 cases of variable C / constant D, constant C / variable D, and variable C / variable D. I am not sure it is worth the extra code to handle all of the cases.

If one of the coefficients is constant,

Reply:
(If one of the coefficients is constant, you can just copy the constant value into a properly sized array.)

Reply:
(OTOH, if you know that it makes a big performance difference, that is a good reason to have different specializations.)

Reply:
We should always strive to reduce memory usage in IBAMR, wherever possible. It will unnecessary create and then destroy temp (stack or heap) memory to hold constant values of C/D.

Reply:
We already have a ton of repeated code, unfortunately, in the FORTRAN routines (e.g., 90% of every interpolation and spread function). It would be great if we could get rid of it but I support the simple solution here since its consistent with what we have and avoids having to allocate lots of extra memory for no reason.